ksqldb-execution/src/test/java/io/confluent/ksql/schema/query/QuerySchemasTest.java

```diff
@@ -0,0 +1,299 @@
+            containsString("Zero key formats registered for topic."));
+        assertThat(e.getMessage(),
+            containsString("topic: topic1"));
+        assertThat(e.getMessage(),
+            containsString("formats: []"));
+        assertThat(e.getMessage(),
+            containsString("loggers: []"));
+    }
+
+    @Test
+    public void shouldThrowIfTopicNameIsNotFound() {
+        // When
+        final Exception e = assertThrows(
+            IllegalArgumentException.class,
+            () -> querySchemas.getTopicFormatsInfo("t1"));
+
+        // Then
+        assertThat(e.getMessage(), is("Unknown topic: t1"));
+    }
+
+    @Test
+    public void shouldThrowIfMultipleKeyLoggersSchemasAreFound() {
+        // Given
+        querySchemas.trackKeySerdeCreation("K_LOGGER_1", LOGICAL_SCHEMA_X, JSON_KEY_FORMAT);
+        querySchemas.trackKeySerdeCreation("K_LOGGER_2", LOGICAL_SCHEMA_X, AVRO_KEY_FORMAT);
+        querySchemas.trackSerdeOp("topic1", IS_KEY, "K_LOGGER_1");
+        querySchemas.trackSerdeOp("topic1", IS_KEY, "K_LOGGER_2");
+
+        // When
+        final Exception e = assertThrows(
+            IllegalStateException.class,
+            () -> querySchemas.getTopicFormatsInfo("topic1"));
+
+        // Then
+        assertThat(e.getMessage(),
+            containsString("Multiple key formats registered for topic."));
+        assertThat(e.getMessage(),
+            containsString("topic: topic1"));
+        assertThat(e.getMessage(),
+            containsString("formats: [AVRO, JSON]"));
+        assertThat(e.getMessage(),
+            containsString("loggers: [[K_LOGGER_1], [K_LOGGER_2]]"));
+    }
+
+    @Test
+    public void shouldThrowIfMultipleValueLoggersSchemasAreFound() {
+        // Given
+        querySchemas.trackKeySerdeCreation("K_LOGGER_1", LOGICAL_SCHEMA_X, JSON_KEY_FORMAT);
+        querySchemas.trackValueSerdeCreation("V_LOGGER_1", LOGICAL_SCHEMA_X, JSON_VALUE_FORMAT);
+        querySchemas.trackValueSerdeCreation("V_LOGGER_2", LOGICAL_SCHEMA_X, AVRO_VALUE_FORMAT);
+        querySchemas.trackSerdeOp("topic1", IS_KEY, "K_LOGGER_1");
+        querySchemas.trackSerdeOp("topic1", IS_VALUE, "V_LOGGER_1");
+        querySchemas.trackSerdeOp("topic1", IS_VALUE, "V_LOGGER_2");
+
+        // When
+        final Exception e = assertThrows(
+            IllegalStateException.class,
+            () -> querySchemas.getTopicFormatsInfo("topic1"));
+
+        // Then
+        assertThat(e.getMessage(),
+            containsString("Multiple value formats registered for topic."));
+        assertThat(e.getMessage(),
+            containsString("topic: topic1"));
+        assertThat(e.getMessage(),
+            containsString("formats: [AVRO, JSON]"));
+        assertThat(e.getMessage(),
+            containsString("loggers: [[V_LOGGER_1], [V_LOGGER_2]"));
+    }
+
+    @Test
+    public void shouldThrowIfKeySchemaIsNotFound() {
+        // Given
+        querySchemas.trackSerdeOp("topic1", IS_KEY, "K_LOGGER_1");
```

Comment:
I don't understand this test. Isn't this a key schema? The test's name suggests that it's testing the case where there are no key schemas, though.

Also, doesn't this test duplicate `shouldThrowFormatsInfoIfNoKeySchemaIsFound()` above?